test/e2e/custom_resource_definition.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/kube-apiextensions-server/pkg/client/clientset/clientset"
+	"k8s.io/kubernetes/pkg/api"
+	//"k8s.io/kube-apiextensions-server/test/integration/testserver"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/davecgh/go-spew/spew"
+)
+
+type CRD struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" description:"standard object metadata"`
+
+	SomeField  string `json:"someField"`
+	OtherField int    `json:"otherField"`
+}
+
+type CRDList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" description:"standard list metadata; see http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata"`
+
+	Items []CRD `json:"items"`
+}
+
+// This test is marked flaky pending namespace controller observing dynamic creation of new third party types.
+var _ = Describe("CustomResourceDefinition resources", func() {
+
+	f := framework.NewDefaultFramework("customresourcedefinition")
+
+	crd := &v1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "noxus.mygroup.example.com",
+		},
+		Spec: v1beta1.CustomResourceDefinitionSpec{
+			Group:   "mygroup.example.com",
+			Version: "v1alpha1",
+			Names: v1beta1.CustomResourceDefinitionNames{
+				Plural:     "noxus",
+				Singular:   "noxu",
+				Kind:       "Noxu",
+				ListKind:   "NoxuList",
+			},
+			Scope: v1beta1.ClusterScoped,
+		},
+	}
+
+	Context("Simple CustomResourceDefinition", func() {
+		It("creating/deleting custom resource definition objects works [Conformance]", func() {
+			config, err := framework.LoadConfig()
```

Comment:
Try actually modifying the real before each here https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/framework.go#L190 and seeing if you get the same problem.